enclave-runtime/src/lib.rs

```diff
@@ -891,12 +890,14 @@ where
 	let mut call_hashes = Vec::<H256>::new();
 
 	// load state before executing any calls
-	let mut state = load_initialized_state(&shard)?;
-	// save the state hash before call executions
-	// (needed for block composition)
-	trace!("Getting hash of previous state ..");
-	let prev_state_hash = state.hash();
-	trace!("Loaded hash of previous state: {:?}", prev_state_hash);
+	let mut sidechain_db = load_initialized_state(&shard).map(SidechainDB::<SB::Block, _>::new)?;
+
+	// update state needed for pallets
+	sidechain_db.set_block_number(&sidechain_db.get_block_number().map_or(1, |n| n + 1));
+	sidechain_db.set_timestamp(&now_as_u64());
+
+	let prev_state_hash = sidechain_db.state_hash();
```

Comment:
hmm, if you derive the state hash after mutating the state, it will not correspond to the a-posteriori state hash of the previous block. is that a problem?

Reply:
You are right. This is wrong.